normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
I wasn't involved in the revision PR so I'm a little behind, but I don't quite get what this method is trying to do. It seems like it's avoiding making another revision if a revision with the same state already exists, but does that mean that we can't have two revisions with the same state, but different parents? Wouldn't this make it impossible to represent a recipe going back to a previous state?